### DISCUSSION TITLE
能否请教一些问题

### DISCUSSION THREAD

**Discussion by benny502** [scarsty/kys-cpp#55] [Q&A] [answered]
最近在研究大大的代码，斜45度地图的渲染这里看的有些头晕。不明白坐标都是如何计算的，还有resource/earth.002里的索引和index.ka里的坐标分别都是什么的索引啊。
我仿照着写了一下，发现surface层和building层都有些错位，一旦移动地图画面又会闪现一些小黑块，实在是头晕脑涨

**Comment by benny502** [comment, marked as answer]
来自问自答吧，earth.002这些文件里的是素材的索引。理论上来讲，靠这里的索引已经能计算出素材的坐标
`    p.x = -y * TILE_W + x * TILE_W + render_center_x_;
    p.y = y * TILE_H + x * TILE_H + render_center_y_;`
但是这些素材不知道为什么还有一定的坐标偏移，存在index.ka里，真正的显示坐标要用计算出来的坐标减去偏移坐标

**Reply by scarsty** [reply]
所有图片的零点定义到下面中间 例如人 建筑等 方便判断物件的地点坐标
